Build 1 for 8.4 with status FAILURE

## :broken_heart: Tests Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.4/detail/8.4/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.4/detail/8.4/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.4/detail/8.4/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.4/detail/8.4/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_null.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2022-07-27T12:54:51.839Z&rangeTo=2022-07-27T13:14:51.839Z&transactionName=Beats/beats/8.4&transactionType=job&latencyAggregationType=avg&traceId=7fd7de1acfa307de48a911893db134df&transactionId=7e9c3a0b469197df)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2022-07-27T13:04:51.839+0000


* Duration: 142 min 2 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 10  |
| Passed       | 24295  |
| Skipped      | 2254 |
| Total        | 26559   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->

  ### Test errors [![10](https://img.shields.io/badge/10%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.4/detail/8.4/1//tests)
<details><summary>Expand to view the tests failures</summary><p>
    
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_dashboards – x-pack.metricbeat.tests.system.test_xpack_base.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <class "test_xpack_base.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class "test_xpack_base.Test">

    @classmethod
    def compose_up(cls):
        """
        Ensure *only* the services defined under `COMPOSE_SERVICES` are running and healthy
        """
        if not INTEGRATION_TESTS or not cls.COMPOSE_SERVICES:
            return
    
        if os.environ.get("NO_COMPOSE"):
            return
    
        def print_logs(container):
            print("---- " + container.name_without_project)
            print(container.logs())
            print("----")
    
        def is_healthy(container):
            return container.inspect()["State"]["Health"]["Status"] == "healthy"
    
        project = cls.compose_project()
    
        with disabled_logger("compose.service"):
            project.pull(
                ignore_pull_failures=True,
                service_names=cls.COMPOSE_SERVICES)
    
        project.up(
            strategy=ConvergenceStrategy.always,
            service_names=cls.COMPOSE_SERVICES,
            timeout=30)
    
        # Wait for them to be healthy
        start = time.time()
        while True:
            containers = project.containers(
                service_names=cls.COMPOSE_SERVICES,
                stopped=True)
    
            healthy = True
            for container in containers:
                if not container.is_running:
                    print_logs(container)
                    raise Exception(
                        "Container %s unexpectedly finished on startup" %
                        container.name_without_project)
                if not is_healthy(container):
                    healthy = False
                    break
    
            if healthy:
                break
    
            if cls.COMPOSE_ADVERTISED_HOST:
                for service in cls.COMPOSE_SERVICES:
                    cls._setup_advertised_host(project, service)
    
            time.sleep(1)
            timeout = time.time() - start > cls.COMPOSE_TIMEOUT
            if timeout:
                for container in containers:
                    if not is_healthy(container):
                        print_logs(container)
>               raise Exception(
                    "Timeout while waiting for healthy "
                    "docker-compose services: %s" %
                    ",".join(cls.COMPOSE_SERVICES))
E               Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana

../../libbeat/tests/system/beat/compose.py:102: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_export_config – x-pack.metricbeat.tests.system.test_xpack_base.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <class "test_xpack_base.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class "test_xpack_base.Test">

    @classmethod
    def compose_up(cls):
        """
        Ensure *only* the services defined under `COMPOSE_SERVICES` are running and healthy
        """
        if not INTEGRATION_TESTS or not cls.COMPOSE_SERVICES:
            return
    
        if os.environ.get("NO_COMPOSE"):
            return
    
        def print_logs(container):
            print("---- " + container.name_without_project)
            print(container.logs())
            print("----")
    
        def is_healthy(container):
            return container.inspect()["State"]["Health"]["Status"] == "healthy"
    
        project = cls.compose_project()
    
        with disabled_logger("compose.service"):
            project.pull(
                ignore_pull_failures=True,
                service_names=cls.COMPOSE_SERVICES)
    
        project.up(
            strategy=ConvergenceStrategy.always,
            service_names=cls.COMPOSE_SERVICES,
            timeout=30)
    
        # Wait for them to be healthy
        start = time.time()
        while True:
            containers = project.containers(
                service_names=cls.COMPOSE_SERVICES,
                stopped=True)
    
            healthy = True
            for container in containers:
                if not container.is_running:
                    print_logs(container)
                    raise Exception(
                        "Container %s unexpectedly finished on startup" %
                        container.name_without_project)
                if not is_healthy(container):
                    healthy = False
                    break
    
            if healthy:
                break
    
            if cls.COMPOSE_ADVERTISED_HOST:
                for service in cls.COMPOSE_SERVICES:
                    cls._setup_advertised_host(project, service)
    
            time.sleep(1)
            timeout = time.time() - start > cls.COMPOSE_TIMEOUT
            if timeout:
                for container in containers:
                    if not is_healthy(container):
                        print_logs(container)
>               raise Exception(
                    "Timeout while waiting for healthy "
                    "docker-compose services: %s" %
                    ",".join(cls.COMPOSE_SERVICES))
E               Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana

../../libbeat/tests/system/beat/compose.py:102: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_export_ilm_policy – x-pack.metricbeat.tests.system.test_xpack_base.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <class "test_xpack_base.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class "test_xpack_base.Test">

    @classmethod
    def compose_up(cls):
        """
        Ensure *only* the services defined under `COMPOSE_SERVICES` are running and healthy
        """
        if not INTEGRATION_TESTS or not cls.COMPOSE_SERVICES:
            return
    
        if os.environ.get("NO_COMPOSE"):
            return
    
        def print_logs(container):
            print("---- " + container.name_without_project)
            print(container.logs())
            print("----")
    
        def is_healthy(container):
            return container.inspect()["State"]["Health"]["Status"] == "healthy"
    
        project = cls.compose_project()
    
        with disabled_logger("compose.service"):
            project.pull(
                ignore_pull_failures=True,
                service_names=cls.COMPOSE_SERVICES)
    
        project.up(
            strategy=ConvergenceStrategy.always,
            service_names=cls.COMPOSE_SERVICES,
            timeout=30)
    
        # Wait for them to be healthy
        start = time.time()
        while True:
            containers = project.containers(
                service_names=cls.COMPOSE_SERVICES,
                stopped=True)
    
            healthy = True
            for container in containers:
                if not container.is_running:
                    print_logs(container)
                    raise Exception(
                        "Container %s unexpectedly finished on startup" %
                        container.name_without_project)
                if not is_healthy(container):
                    healthy = False
                    break
    
            if healthy:
                break
    
            if cls.COMPOSE_ADVERTISED_HOST:
                for service in cls.COMPOSE_SERVICES:
                    cls._setup_advertised_host(project, service)
    
            time.sleep(1)
            timeout = time.time() - start > cls.COMPOSE_TIMEOUT
            if timeout:
                for container in containers:
                    if not is_healthy(container):
                        print_logs(container)
>               raise Exception(
                    "Timeout while waiting for healthy "
                    "docker-compose services: %s" %
                    ",".join(cls.COMPOSE_SERVICES))
E               Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana

../../libbeat/tests/system/beat/compose.py:102: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_export_index_pattern – x-pack.metricbeat.tests.system.test_xpack_base.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <class "test_xpack_base.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class "test_xpack_base.Test">

    @classmethod
    def compose_up(cls):
        """
        Ensure *only* the services defined under `COMPOSE_SERVICES` are running and healthy
        """
        if not INTEGRATION_TESTS or not cls.COMPOSE_SERVICES:
            return
    
        if os.environ.get("NO_COMPOSE"):
            return
    
        def print_logs(container):
            print("---- " + container.name_without_project)
            print(container.logs())
            print("----")
    
        def is_healthy(container):
            return container.inspect()["State"]["Health"]["Status"] == "healthy"
    
        project = cls.compose_project()
    
        with disabled_logger("compose.service"):
            project.pull(
                ignore_pull_failures=True,
                service_names=cls.COMPOSE_SERVICES)
    
        project.up(
            strategy=ConvergenceStrategy.always,
            service_names=cls.COMPOSE_SERVICES,
            timeout=30)
    
        # Wait for them to be healthy
        start = time.time()
        while True:
            containers = project.containers(
                service_names=cls.COMPOSE_SERVICES,
                stopped=True)
    
            healthy = True
            for container in containers:
                if not container.is_running:
                    print_logs(container)
                    raise Exception(
                        "Container %s unexpectedly finished on startup" %
                        container.name_without_project)
                if not is_healthy(container):
                    healthy = False
                    break
    
            if healthy:
                break
    
            if cls.COMPOSE_ADVERTISED_HOST:
                for service in cls.COMPOSE_SERVICES:
                    cls._setup_advertised_host(project, service)
    
            time.sleep(1)
            timeout = time.time() - start > cls.COMPOSE_TIMEOUT
            if timeout:
                for container in containers:
                    if not is_healthy(container):
                        print_logs(container)
>               raise Exception(
                    "Timeout while waiting for healthy "
                    "docker-compose services: %s" %
                    ",".join(cls.COMPOSE_SERVICES))
E               Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana

../../libbeat/tests/system/beat/compose.py:102: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_export_index_pattern_migration – x-pack.metricbeat.tests.system.test_xpack_base.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <class "test_xpack_base.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class "test_xpack_base.Test">

    @classmethod
    def compose_up(cls):
        """
        Ensure *only* the services defined under `COMPOSE_SERVICES` are running and healthy
        """
        if not INTEGRATION_TESTS or not cls.COMPOSE_SERVICES:
            return
    
        if os.environ.get("NO_COMPOSE"):
            return
    
        def print_logs(container):
            print("---- " + container.name_without_project)
            print(container.logs())
            print("----")
    
        def is_healthy(container):
            return container.inspect()["State"]["Health"]["Status"] == "healthy"
    
        project = cls.compose_project()
    
        with disabled_logger("compose.service"):
            project.pull(
                ignore_pull_failures=True,
                service_names=cls.COMPOSE_SERVICES)
    
        project.up(
            strategy=ConvergenceStrategy.always,
            service_names=cls.COMPOSE_SERVICES,
            timeout=30)
    
        # Wait for them to be healthy
        start = time.time()
        while True:
            containers = project.containers(
                service_names=cls.COMPOSE_SERVICES,
                stopped=True)
    
            healthy = True
            for container in containers:
                if not container.is_running:
                    print_logs(container)
                    raise Exception(
                        "Container %s unexpectedly finished on startup" %
                        container.name_without_project)
                if not is_healthy(container):
                    healthy = False
                    break
    
            if healthy:
                break
    
            if cls.COMPOSE_ADVERTISED_HOST:
                for service in cls.COMPOSE_SERVICES:
                    cls._setup_advertised_host(project, service)
    
            time.sleep(1)
            timeout = time.time() - start > cls.COMPOSE_TIMEOUT
            if timeout:
                for container in containers:
                    if not is_healthy(container):
                        print_logs(container)
>               raise Exception(
                    "Timeout while waiting for healthy "
                    "docker-compose services: %s" %
                    ",".join(cls.COMPOSE_SERVICES))
E               Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana

../../libbeat/tests/system/beat/compose.py:102: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_export_template – x-pack.metricbeat.tests.system.test_xpack_base.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <class "test_xpack_base.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class "test_xpack_base.Test">

    @classmethod
    def compose_up(cls):
        """
        Ensure *only* the services defined under `COMPOSE_SERVICES` are running and healthy
        """
        if not INTEGRATION_TESTS or not cls.COMPOSE_SERVICES:
            return
    
        if os.environ.get("NO_COMPOSE"):
            return
    
        def print_logs(container):
            print("---- " + container.name_without_project)
            print(container.logs())
            print("----")
    
        def is_healthy(container):
            return container.inspect()["State"]["Health"]["Status"] == "healthy"
    
        project = cls.compose_project()
    
        with disabled_logger("compose.service"):
            project.pull(
                ignore_pull_failures=True,
                service_names=cls.COMPOSE_SERVICES)
    
        project.up(
            strategy=ConvergenceStrategy.always,
            service_names=cls.COMPOSE_SERVICES,
            timeout=30)
    
        # Wait for them to be healthy
        start = time.time()
        while True:
            containers = project.containers(
                service_names=cls.COMPOSE_SERVICES,
                stopped=True)
    
            healthy = True
            for container in containers:
                if not container.is_running:
                    print_logs(container)
                    raise Exception(
                        "Container %s unexpectedly finished on startup" %
                        container.name_without_project)
                if not is_healthy(container):
                    healthy = False
                    break
    
            if healthy:
                break
    
            if cls.COMPOSE_ADVERTISED_HOST:
                for service in cls.COMPOSE_SERVICES:
                    cls._setup_advertised_host(project, service)
    
            time.sleep(1)
            timeout = time.time() - start > cls.COMPOSE_TIMEOUT
            if timeout:
                for container in containers:
                    if not is_healthy(container):
                        print_logs(container)
>               raise Exception(
                    "Timeout while waiting for healthy "
                    "docker-compose services: %s" %
                    ",".join(cls.COMPOSE_SERVICES))
E               Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana

../../libbeat/tests/system/beat/compose.py:102: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_index_management – x-pack.metricbeat.tests.system.test_xpack_base.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <class "test_xpack_base.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class "test_xpack_base.Test">

    @classmethod
    def compose_up(cls):
        """
        Ensure *only* the services defined under `COMPOSE_SERVICES` are running and healthy
        """
        if not INTEGRATION_TESTS or not cls.COMPOSE_SERVICES:
            return
    
        if os.environ.get("NO_COMPOSE"):
            return
    
        def print_logs(container):
            print("---- " + container.name_without_project)
            print(container.logs())
            print("----")
    
        def is_healthy(container):
            return container.inspect()["State"]["Health"]["Status"] == "healthy"
    
        project = cls.compose_project()
    
        with disabled_logger("compose.service"):
            project.pull(
                ignore_pull_failures=True,
                service_names=cls.COMPOSE_SERVICES)
    
        project.up(
            strategy=ConvergenceStrategy.always,
            service_names=cls.COMPOSE_SERVICES,
            timeout=30)
    
        # Wait for them to be healthy
        start = time.time()
        while True:
            containers = project.containers(
                service_names=cls.COMPOSE_SERVICES,
                stopped=True)
    
            healthy = True
            for container in containers:
                if not container.is_running:
                    print_logs(container)
                    raise Exception(
                        "Container %s unexpectedly finished on startup" %
                        container.name_without_project)
                if not is_healthy(container):
                    healthy = False
                    break
    
            if healthy:
                break
    
            if cls.COMPOSE_ADVERTISED_HOST:
                for service in cls.COMPOSE_SERVICES:
                    cls._setup_advertised_host(project, service)
    
            time.sleep(1)
            timeout = time.time() - start > cls.COMPOSE_TIMEOUT
            if timeout:
                for container in containers:
                    if not is_healthy(container):
                        print_logs(container)
>               raise Exception(
                    "Timeout while waiting for healthy "
                    "docker-compose services: %s" %
                    ",".join(cls.COMPOSE_SERVICES))
E               Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana

../../libbeat/tests/system/beat/compose.py:102: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_start_stop – x-pack.metricbeat.tests.system.test_xpack_base.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <class "test_xpack_base.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class "test_xpack_base.Test">

    @classmethod
    def compose_up(cls):
        """
        Ensure *only* the services defined under `COMPOSE_SERVICES` are running and healthy
        """
        if not INTEGRATION_TESTS or not cls.COMPOSE_SERVICES:
            return
    
        if os.environ.get("NO_COMPOSE"):
            return
    
        def print_logs(container):
            print("---- " + container.name_without_project)
            print(container.logs())
            print("----")
    
        def is_healthy(container):
            return container.inspect()["State"]["Health"]["Status"] == "healthy"
    
        project = cls.compose_project()
    
        with disabled_logger("compose.service"):
            project.pull(
                ignore_pull_failures=True,
                service_names=cls.COMPOSE_SERVICES)
    
        project.up(
            strategy=ConvergenceStrategy.always,
            service_names=cls.COMPOSE_SERVICES,
            timeout=30)
    
        # Wait for them to be healthy
        start = time.time()
        while True:
            containers = project.containers(
                service_names=cls.COMPOSE_SERVICES,
                stopped=True)
    
            healthy = True
            for container in containers:
                if not container.is_running:
                    print_logs(container)
                    raise Exception(
                        "Container %s unexpectedly finished on startup" %
                        container.name_without_project)
                if not is_healthy(container):
                    healthy = False
                    break
    
            if healthy:
                break
    
            if cls.COMPOSE_ADVERTISED_HOST:
                for service in cls.COMPOSE_SERVICES:
                    cls._setup_advertised_host(project, service)
    
            time.sleep(1)
            timeout = time.time() - start > cls.COMPOSE_TIMEOUT
            if timeout:
                for container in containers:
                    if not is_healthy(container):
                        print_logs(container)
>               raise Exception(
                    "Timeout while waiting for healthy "
                    "docker-compose services: %s" %
                    ",".join(cls.COMPOSE_SERVICES))
E               Exception: Timeout while waiting for healthy docker-compose services: elasticsearch,kibana

../../libbeat/tests/system/beat/compose.py:102: Exception 
 ```
</p></details>
</ul>
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_health – x-pack.metricbeat.module.enterprisesearch.test_enterprisesearch.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "DeprecationWarning: The "warn" method is deprecated, use "warning" instead" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <docker.api.client.APIClient object at 0x7fddfc6279d0>
response = <Response [500]>

    def _raise_for_status(self, response):
        """Raises stored :class:`APIError`, if one occurred."""
        try:
>           response.raise_for_status()

../../build/ve/docker/lib/python3.9/site-packages/docker/api/client.py:268: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Response [500]>

    def raise_for_status(self):
        """Raises :class:`HTTPError`, if one occurred."""
    
        http_error_msg = ""
        if isinstance(self.reason, bytes):
            # We attempt to decode utf-8 first because some servers
            # choose to localize their reason strings. If the string
            # isn"t utf-8, we fall back to iso-8859-1 for all other
            # encodings. (See PR #3538)
            try:
                reason = self.reason.decode("utf-8")
            except UnicodeDecodeError:
                reason = self.reason.decode("iso-8859-1")
        else:
            reason = self.reason
    
        if 400 <= self.status_code < 500:
            http_error_msg = u"%s Client Error: %s for url: %s" % (self.status_code, reason, self.url)
    
        elif 500 <= self.status_code < 600:
            http_error_msg = u"%s Server Error: %s for url: %s" % (self.status_code, reason, self.url)
    
        if http_error_msg:
>           raise HTTPError(http_error_msg, response=self)
E           requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http+docker://localhost/v1.41/containers/3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085/start

../../build/ve/docker/lib/python3.9/site-packages/requests/models.py:943: HTTPError

During handling of the above exception, another exception occurred:

self = <Service: elasticsearch>
container = <Container: enterprisesearch_7918a246f6f8_elasticsearch_1 (3be66e)>
use_network_aliases = True

    def start_container(self, container, use_network_aliases=True):
        self.connect_container_to_networks(container, use_network_aliases)
        try:
>           container.start()

../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:643: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Container: enterprisesearch_7918a246f6f8_elasticsearch_1 (3be66e)>
options = {}

    def start(self, **options):
>       return self.client.start(self.id, **options)

../../build/ve/docker/lib/python3.9/site-packages/compose/container.py:228: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <docker.api.client.APIClient object at 0x7fddfc6279d0>
resource_id = "3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085"
args = (), kwargs = {}

    @functools.wraps(f)
    def wrapped(self, resource_id=None, *args, **kwargs):
        if resource_id is None and kwargs.get(resource_name):
            resource_id = kwargs.pop(resource_name)
        if isinstance(resource_id, dict):
            resource_id = resource_id.get("Id", resource_id.get("ID"))
        if not resource_id:
            raise errors.NullResource(
                "Resource ID was not provided"
            )
>       return f(self, resource_id, *args, **kwargs)

../../build/ve/docker/lib/python3.9/site-packages/docker/utils/decorators.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <docker.api.client.APIClient object at 0x7fddfc6279d0>
container = "3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085"
args = (), kwargs = {}
url = "http+docker://localhost/v1.41/containers/3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085/start"
res = <Response [500]>

    @utils.check_resource("container")
    def start(self, container, *args, **kwargs):
        """
        Start a container. Similar to the ``docker start`` command, but
        doesn"t support attach options.
    
        **Deprecation warning:** Passing configuration options in ``start`` is
        no longer supported. Users are expected to provide host config options
        in the ``host_config`` parameter of
        :py:meth:`~ContainerApiMixin.create_container`.
    
    
        Args:
            container (str): The container to start
    
        Raises:
            :py:class:`docker.errors.APIError`
                If the server returns an error.
            :py:class:`docker.errors.DeprecatedMethod`
                If any argument besides ``container`` are provided.
    
        Example:
    
            >>> container = client.api.create_container(
            ...     image="busybox:latest",
            ...     command="/bin/sleep 30")
            >>> client.api.start(container=container.get("Id"))
        """
        if args or kwargs:
            raise errors.DeprecatedMethod(
                "Providing configuration in the start() method is no longer "
                "supported. Use the host_config param in create_container "
                "instead."
            )
        url = self._url("/containers/{0}/start", container)
        res = self._post(url)
>       self._raise_for_status(res)

../../build/ve/docker/lib/python3.9/site-packages/docker/api/container.py:1109: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <docker.api.client.APIClient object at 0x7fddfc6279d0>
response = <Response [500]>

    def _raise_for_status(self, response):
        """Raises stored :class:`APIError`, if one occurred."""
        try:
            response.raise_for_status()
        except requests.exceptions.HTTPError as e:
>           raise create_api_error_from_http_exception(e)

../../build/ve/docker/lib/python3.9/site-packages/docker/api/client.py:270: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

e = HTTPError("500 Server Error: Internal Server Error for url: http+docker://localhost/v1.41/containers/3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085/start")

    def create_api_error_from_http_exception(e):
        """
        Create a suitable APIError from requests.exceptions.HTTPError.
        """
        response = e.response
        try:
            explanation = response.json()["message"]
        except ValueError:
            explanation = (response.content or "").strip()
        cls = APIError
        if response.status_code == 404:
            if explanation and ("No such image" in str(explanation) or
                                "not found: does not exist or no pull access"
                                in str(explanation) or
                                "repository does not exist" in str(explanation)):
                cls = ImageNotFound
            else:
                cls = NotFound
>       raise cls(e, response=response, explanation=explanation)
E       docker.errors.APIError: 500 Server Error for http+docker://localhost/v1.41/containers/3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085/start: Internal Server Error ("driver failed programming external connectivity on endpoint enterprisesearch_7918a246f6f8_elasticsearch_1 (893babbc33811fcaaea7b4acf8fd3a1e49b6ce9023b3db42ab515ea6861cfa81): Bind for 0.0.0.0:9200 failed: port is already allocated")

../../build/ve/docker/lib/python3.9/site-packages/docker/errors.py:31: APIError

During handling of the above exception, another exception occurred:

self = <class "test_enterprisesearch.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
../../libbeat/tests/system/beat/compose.py:66: in compose_up
    project.up(
../../build/ve/docker/lib/python3.9/site-packages/compose/project.py:697: in up
    results, errors = parallel.parallel_execute(
../../build/ve/docker/lib/python3.9/site-packages/compose/parallel.py:108: in parallel_execute
    raise error_to_reraise
../../build/ve/docker/lib/python3.9/site-packages/compose/parallel.py:206: in producer
    result = func(obj)
../../build/ve/docker/lib/python3.9/site-packages/compose/project.py:679: in do
    return service.execute_convergence_plan(
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:559: in execute_convergence_plan
    return self._execute_convergence_create(
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:473: in _execute_convergence_create
    containers, errors = parallel_execute(
../../build/ve/docker/lib/python3.9/site-packages/compose/parallel.py:108: in parallel_execute
    raise error_to_reraise
../../build/ve/docker/lib/python3.9/site-packages/compose/parallel.py:206: in producer
    result = func(obj)
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:478: in <lambda>
    lambda service_name: create_and_start(self, service_name.number),
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:461: in create_and_start
    self.start_container(container)
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:647: in start_container
    log.warn("Host is already in use by another container")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Logger compose.service (WARNING)>
msg = "Host is already in use by another container", args = (), kwargs = {}

    def warn(self, msg, *args, **kwargs):
>       warnings.warn("The "warn" method is deprecated, "
            "use "warning" instead", DeprecationWarning, 2)
E       DeprecationWarning: The "warn" method is deprecated, use "warning" instead

/usr/lib/python3.9/logging/__init__.py:1457: DeprecationWarning 
 ```
</p></details>
</ul>
    
##### `Build&Test / x-pack/metricbeat-pythonIntegTest / test_stats – x-pack.metricbeat.module.enterprisesearch.test_enterprisesearch.Test`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 failed on setup with "DeprecationWarning: The "warn" method is deprecated, use "warning" instead" 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 self = <docker.api.client.APIClient object at 0x7fddfc6279d0>
response = <Response [500]>

    def _raise_for_status(self, response):
        """Raises stored :class:`APIError`, if one occurred."""
        try:
>           response.raise_for_status()

../../build/ve/docker/lib/python3.9/site-packages/docker/api/client.py:268: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Response [500]>

    def raise_for_status(self):
        """Raises :class:`HTTPError`, if one occurred."""
    
        http_error_msg = ""
        if isinstance(self.reason, bytes):
            # We attempt to decode utf-8 first because some servers
            # choose to localize their reason strings. If the string
            # isn"t utf-8, we fall back to iso-8859-1 for all other
            # encodings. (See PR #3538)
            try:
                reason = self.reason.decode("utf-8")
            except UnicodeDecodeError:
                reason = self.reason.decode("iso-8859-1")
        else:
            reason = self.reason
    
        if 400 <= self.status_code < 500:
            http_error_msg = u"%s Client Error: %s for url: %s" % (self.status_code, reason, self.url)
    
        elif 500 <= self.status_code < 600:
            http_error_msg = u"%s Server Error: %s for url: %s" % (self.status_code, reason, self.url)
    
        if http_error_msg:
>           raise HTTPError(http_error_msg, response=self)
E           requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http+docker://localhost/v1.41/containers/3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085/start

../../build/ve/docker/lib/python3.9/site-packages/requests/models.py:943: HTTPError

During handling of the above exception, another exception occurred:

self = <Service: elasticsearch>
container = <Container: enterprisesearch_7918a246f6f8_elasticsearch_1 (3be66e)>
use_network_aliases = True

    def start_container(self, container, use_network_aliases=True):
        self.connect_container_to_networks(container, use_network_aliases)
        try:
>           container.start()

../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:643: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Container: enterprisesearch_7918a246f6f8_elasticsearch_1 (3be66e)>
options = {}

    def start(self, **options):
>       return self.client.start(self.id, **options)

../../build/ve/docker/lib/python3.9/site-packages/compose/container.py:228: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <docker.api.client.APIClient object at 0x7fddfc6279d0>
resource_id = "3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085"
args = (), kwargs = {}

    @functools.wraps(f)
    def wrapped(self, resource_id=None, *args, **kwargs):
        if resource_id is None and kwargs.get(resource_name):
            resource_id = kwargs.pop(resource_name)
        if isinstance(resource_id, dict):
            resource_id = resource_id.get("Id", resource_id.get("ID"))
        if not resource_id:
            raise errors.NullResource(
                "Resource ID was not provided"
            )
>       return f(self, resource_id, *args, **kwargs)

../../build/ve/docker/lib/python3.9/site-packages/docker/utils/decorators.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <docker.api.client.APIClient object at 0x7fddfc6279d0>
container = "3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085"
args = (), kwargs = {}
url = "http+docker://localhost/v1.41/containers/3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085/start"
res = <Response [500]>

    @utils.check_resource("container")
    def start(self, container, *args, **kwargs):
        """
        Start a container. Similar to the ``docker start`` command, but
        doesn"t support attach options.
    
        **Deprecation warning:** Passing configuration options in ``start`` is
        no longer supported. Users are expected to provide host config options
        in the ``host_config`` parameter of
        :py:meth:`~ContainerApiMixin.create_container`.
    
    
        Args:
            container (str): The container to start
    
        Raises:
            :py:class:`docker.errors.APIError`
                If the server returns an error.
            :py:class:`docker.errors.DeprecatedMethod`
                If any argument besides ``container`` are provided.
    
        Example:
    
            >>> container = client.api.create_container(
            ...     image="busybox:latest",
            ...     command="/bin/sleep 30")
            >>> client.api.start(container=container.get("Id"))
        """
        if args or kwargs:
            raise errors.DeprecatedMethod(
                "Providing configuration in the start() method is no longer "
                "supported. Use the host_config param in create_container "
                "instead."
            )
        url = self._url("/containers/{0}/start", container)
        res = self._post(url)
>       self._raise_for_status(res)

../../build/ve/docker/lib/python3.9/site-packages/docker/api/container.py:1109: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <docker.api.client.APIClient object at 0x7fddfc6279d0>
response = <Response [500]>

    def _raise_for_status(self, response):
        """Raises stored :class:`APIError`, if one occurred."""
        try:
            response.raise_for_status()
        except requests.exceptions.HTTPError as e:
>           raise create_api_error_from_http_exception(e)

../../build/ve/docker/lib/python3.9/site-packages/docker/api/client.py:270: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

e = HTTPError("500 Server Error: Internal Server Error for url: http+docker://localhost/v1.41/containers/3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085/start")

    def create_api_error_from_http_exception(e):
        """
        Create a suitable APIError from requests.exceptions.HTTPError.
        """
        response = e.response
        try:
            explanation = response.json()["message"]
        except ValueError:
            explanation = (response.content or "").strip()
        cls = APIError
        if response.status_code == 404:
            if explanation and ("No such image" in str(explanation) or
                                "not found: does not exist or no pull access"
                                in str(explanation) or
                                "repository does not exist" in str(explanation)):
                cls = ImageNotFound
            else:
                cls = NotFound
>       raise cls(e, response=response, explanation=explanation)
E       docker.errors.APIError: 500 Server Error for http+docker://localhost/v1.41/containers/3be66e09f0067e0e3200934977c0d350ec3a1a9a64a7028676a3e2154d88d085/start: Internal Server Error ("driver failed programming external connectivity on endpoint enterprisesearch_7918a246f6f8_elasticsearch_1 (893babbc33811fcaaea7b4acf8fd3a1e49b6ce9023b3db42ab515ea6861cfa81): Bind for 0.0.0.0:9200 failed: port is already allocated")

../../build/ve/docker/lib/python3.9/site-packages/docker/errors.py:31: APIError

During handling of the above exception, another exception occurred:

self = <class "test_enterprisesearch.Test">

    @classmethod
    def setUpClass(self):
        self.beat_name = "metricbeat"
        self.beat_path = os.path.abspath(
            os.path.join(os.path.dirname(__file__), "../../"))
    
        self.template_paths = [
            os.path.abspath(os.path.join(self.beat_path, "../../metricbeat")),
            os.path.abspath(os.path.join(self.beat_path, "../../libbeat")),
        ]
    
>       super(XPackTest, self).setUpClass()

tests/system/xpack_metricbeat.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metricbeat/tests/system/metricbeat.py:42: in setUpClass
    super().setUpClass()
../../libbeat/tests/system/beat/beat.py:204: in setUpClass
    cls.compose_up_with_retries()
../../libbeat/tests/system/beat/beat.py:222: in compose_up_with_retries
    raise ex
../../libbeat/tests/system/beat/beat.py:218: in compose_up_with_retries
    cls.compose_up()
../../libbeat/tests/system/beat/compose.py:66: in compose_up
    project.up(
../../build/ve/docker/lib/python3.9/site-packages/compose/project.py:697: in up
    results, errors = parallel.parallel_execute(
../../build/ve/docker/lib/python3.9/site-packages/compose/parallel.py:108: in parallel_execute
    raise error_to_reraise
../../build/ve/docker/lib/python3.9/site-packages/compose/parallel.py:206: in producer
    result = func(obj)
../../build/ve/docker/lib/python3.9/site-packages/compose/project.py:679: in do
    return service.execute_convergence_plan(
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:559: in execute_convergence_plan
    return self._execute_convergence_create(
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:473: in _execute_convergence_create
    containers, errors = parallel_execute(
../../build/ve/docker/lib/python3.9/site-packages/compose/parallel.py:108: in parallel_execute
    raise error_to_reraise
../../build/ve/docker/lib/python3.9/site-packages/compose/parallel.py:206: in producer
    result = func(obj)
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:478: in <lambda>
    lambda service_name: create_and_start(self, service_name.number),
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:461: in create_and_start
    self.start_container(container)
../../build/ve/docker/lib/python3.9/site-packages/compose/service.py:647: in start_container
    log.warn("Host is already in use by another container")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Logger compose.service (WARNING)>
msg = "Host is already in use by another container", args = (), kwargs = {}

    def warn(self, msg, *args, **kwargs):
>       warnings.warn("The "warn" method is deprecated, "
            "use "warning" instead", DeprecationWarning, 2)
E       DeprecationWarning: The "warn" method is deprecated, use "warning" instead

/usr/lib/python3.9/logging/__init__.py:1457: DeprecationWarning 
 ```
</p></details>
</ul>
    
</p></details>


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![5](https://img.shields.io/badge/5%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.4/detail/8.4/1//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `metricbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 30 min 7 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.4/runs/1/steps/19316/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `x-pack/metricbeat-pythonIntegTest - mage pythonIntegTest`
<ul>
<li>Took 26 min 17 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.4/runs/1/steps/18475/log/?start=0">here</a></li>
<li>Description: <code>mage pythonIntegTest</code></l1>
</ul>
  
  
  
  
##### `x-pack/metricbeat-pythonIntegTest - mage pythonIntegTest`
<ul>
<li>Took 20 min 38 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.4/runs/1/steps/23753/log/?start=0">here</a></li>
<li>Description: <code>mage pythonIntegTest</code></l1>
</ul>
  
  
  
  
##### `x-pack/metricbeat-pythonIntegTest - mage pythonIntegTest`
<ul>
<li>Took 19 min 5 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.4/runs/1/steps/24193/log/?start=0">here</a></li>
<li>Description: <code>mage pythonIntegTest</code></l1>
</ul>
  
  
  
  
##### `Error signal`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.4/runs/1/steps/24308/log/?start=0">here</a></li>
<li>Description: <code>Error "hudson.AbortException: script returned exit code 1"</code></l1>
</ul>
  
  </p>
  </details>

--- a/metricbeat/docker-compose.yml
+++ b/metricbeat/docker-compose.yml
@@ -17,11 +17,11 @@ services:
 
   # Used by base tests
   elasticsearch:
-    image: docker.elastic.co/integrations-ci/beats-elasticsearch:${ELASTICSEARCH_VERSION:-8.1.2}-1
+    image: docker.elastic.co/integrations-ci/beats-elasticsearch:${ELASTICSEARCH_VERSION:-8.3.2}-1
     build:
       context: ./module/elasticsearch/_meta
       args:
-        ELASTICSEARCH_VERSION: ${ELASTICSEARCH_VERSION:-8.1.2}
+        ELASTICSEARCH_VERSION: ${ELASTICSEARCH_VERSION:-8.3.2}
     environment:
       - "ES_JAVA_OPTS=-Xms256m -Xmx256m"
       - "transport.host=127.0.0.1"
@@ -38,11 +38,11 @@ services:
 
   # Used by base tests
   kibana:
-    image: docker.elastic.co/integrations-ci/beats-kibana:${KIBANA_VERSION:-8.1.2}-1
+    image: docker.elastic.co/integrations-ci/beats-kibana:${KIBANA_VERSION:-8.3.2}-1
     build:
       context: ./module/kibana/_meta
       args:
-        KIBANA_VERSION: ${KIBANA_VERSION:-8.1.2}
+        KIBANA_VERSION: ${KIBANA_VERSION:-8.3.2}
     healthcheck:
       test: ["CMD-SHELL", "curl -u beats:testing -s http://localhost:5601/api/status?v8format=true | grep -q '\"overall\":{\"level\":\"available\"'"]
       retries: 600
@@ -53,11 +53,11 @@ services:
 
   # Used by base tests
   metricbeat:
-    image: docker.elastic.co/integrations-ci/beats-metricbeat:${BEAT_VERSION:-8.1.2}-1
+    image: docker.elastic.co/integrations-ci/beats-metricbeat:${BEAT_VERSION:-8.3.2}-1
     build:
       context: ./module/beat/_meta
       args:
-        BEAT_VERSION: ${BEAT_VERSION:-8.1.2}
+        BEAT_VERSION: ${BEAT_VERSION:-8.3.2}
     command: '-e'
     ports:
       - 5066:5066

--- a/testing/environments/snapshot.yml
+++ b/testing/environments/snapshot.yml
@@ -3,7 +3,7 @@
 version: '2.3'
 services:
   elasticsearch:
-    image: docker.elastic.co/elasticsearch/elasticsearch:8.4.0-31269fd2-SNAPSHOT
+    image: docker.elastic.co/elasticsearch/elasticsearch:8.4.0-5954f776-SNAPSHOT
     # When extend is used it merges healthcheck.tests, see:
     # https://github.com/docker/compose/issues/8962
     # healthcheck:
@@ -31,7 +31,7 @@ services:
     - "./docker/elasticsearch/users_roles:/usr/share/elasticsearch/config/users_roles"
 
   logstash:
-    image: docker.elastic.co/logstash/logstash:8.4.0-31269fd2-SNAPSHOT
+    image: docker.elastic.co/logstash/logstash:8.4.0-5954f776-SNAPSHOT
     healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:9600/_node/stats"]
       retries: 600
@@ -44,7 +44,7 @@ services:
       - 5055:5055
 
   kibana:
-    image: docker.elastic.co/kibana/kibana:8.4.0-31269fd2-SNAPSHOT
+    image: docker.elastic.co/kibana/kibana:8.4.0-5954f776-SNAPSHOT
     environment:
     - "ELASTICSEARCH_USERNAME=kibana_system_user"
     - "ELASTICSEARCH_PASSWORD=testing"

